enterprise_access/apps/bffs/api.py

```diff
@@ -96,6 +103,43 @@ def get_and_cache_enterprise_customer(
     return response_payload
 
 
+def get_and_cache_secured_algolia_search_keys(
+    request,
+    enterprise_customer_uuid,
+    timeout=settings.DEFAULT_CACHE_TIMEOUT,
+):
+    """
+    Retrieves and caches default enterprise enrollment intentions for a learner.
+    This function does not cache this data if it includes any enrollable intentions,
+    because we don't want to re-attempt enrollment realization on the second of consecutive requests.
+    """
+    cache_key = secured_algolia_api_key_cache_key(
+        enterprise_customer_uuid,
+        request.user.id,
+    )
+    cached_response = TieredCache.get_cached_response(cache_key)
+    if cached_response.is_found:
+        logger.info(
+            f'secured_algolia_api_key cache hit '
+            f'for enterprise_customer_uuid {enterprise_customer_uuid}'
+            f'and user id {request.user.id}'
+        )
+        return cached_response.value
+
+    client = EnterpriseCatalogUserV1ApiClient(request)
+    response_payload = client.get_secured_algolia_api_key(
+        enterprise_customer_uuid=enterprise_customer_uuid,
+    )
+
+    # Calculate timeout based on response_payload's valid_until field, it should be the
+    # valid until - the smallest unit of time
+    algolia_response = response_payload.get('algolia', {})
+    if valid_until_date := algolia_response.get('valid_until'):
+        timeout = time_until(valid_until_date, 'seconds') - EPSILON_SECONDS
```

Comment:
we probably want a larger buffer here, maybe 10 seconds?  You could also define `epsilon_seconds` as a simple variable in this function - it's just a little buffer we're building in, it's unlikely that we'd ever have to adjust it.